README.md

```diff
@@ -1,38 +1,343 @@
+banned (and by whom) and when an Auto mod event is triggered.
+
+In order for this module to work, it must be both enabled AND have the `LOG_CHANNEL` global setting populated with the
+ID of a channel
+
+This is a quick non-admin style logger that can log:
+
+- Member join
+- Member kick
+- Member leave
+- Member muted
+- Member un-muted
+- Member ban
+
+***
+
+#### Admin Loggers
+
+Admin Loggers are an embed style logger that will post realtime, detailed events to a chosen channel, these events are
+presented in a rich-embed. All triggered events contain an executor, in that will tell you who did what.
+
+for example, if a message is deleted, or a member is kicked or banned, or even if a channel was modified or created. It
+will query your servers event log and pull the person who did that action. so it will tell you who deleted the message,
+kicked or banned the user or deleted the channel.
+
+In order for these modules to work, it must be both enabled AND have the `ADMIN_LOG_CHANNEL` global setting populated
+with the ID of a channel.
+
+To enable this, you must have a
+
+they consist of:
+
+* Channel Logger
+* Guild Logger
+* Member Logger
+* Message Logger
+* Role Logger
+
+##### Channel Logger
+
+The Channel logger will log:
+
+- Channel Create
+- Channel Delete
+- Channel Updated
+- Thread Create
+- Thread Delete
+- Thread Update
+
+##### Guild Logger
+
+The Guild logger will log:
+
+- Server changes
+  - name
+  - description
+  - banner
+  - rules channel (community)
+  - splash image
+  - Discovery image (community)
+  - Server Icon
+  - Vanity URL
+
+##### Member Logger
+
+The Member logger will log:
+
+* Member join
+* Member ban
+* Member kick
+* Member leave
+* Member unBanned
+* Member updated
+* Member joins/leaves VC
+
+##### Message Logger
+
+The Message logger will log:
+
+* Message Edited
+* Messaged Deleted
+* Bulk Message Deletion
+
+Note for Message deleted. this will not only log the deleted message, but will log (and post), any attachment that was
+contained in the message.
+
+so if a message was deleted, and it contained an image, that image will be logged. if it contained a video, the video
+will be logged. this will work for ANY attachment, be it image, video, binary file, a pdf, etc...
+
+##### Role Logger
+
+The Role logger will log:
+
+* Role Created
+* Role Deleted
+* Role Updated
+* Role Given
+* Role Removed
+
+***
+
+#### AutoMod
+
+The auto mod module is a power moderation filter that augments the built-in discord auto mod feature with extra features
+such as kicking, muting, banning and warning users, logging them and using regex in pattern matching for text.
+
+this module relies on a number of submodules known as filters. a Filter is a self-contained and isolated unit of work
+that takes a message and returns true or false if that message violated the filter.
+
+Each filter can be turned off and on
```

Comment:
Sounds stupid but switch off and on around as on and off sounds a lot better than off and on